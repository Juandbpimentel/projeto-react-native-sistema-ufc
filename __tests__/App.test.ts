import "@testing-library/jest-native/extend-expect";
import { render } from "@testing-library/react-native";
import MeuComponent from "../App";

describe("Suíte de Testes para o componente App", () => {
  it("O componente deve estar definido corretamente", () => {
    const componente = render(MeuComponent());

    expect(componente).toBeDefined();
  });

  it("O componente deve estar com a estrutura definida corretamente", () => {
    const componente = render(MeuComponent());

    expect(componente.toJSON()).toMatchSnapshot();
  });

  it("O Componente deve conter um texto específico na tela", () => {
    const { getByText } = render(MeuComponent());
    const componente = getByText("This is my first app");

    expect(componente).toHaveTextContent("This is my first app");
  });
});
