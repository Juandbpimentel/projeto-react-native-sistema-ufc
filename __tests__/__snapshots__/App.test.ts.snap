// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Su√≠te de Testes para o componente App O componente deve estar com a estrutura definida corretamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    id="title"
    style={
      {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    This is my first app
  </Text>
  <View
    style={
      {
        "flex": 1,
        "paddingTop": 28,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "borderRadius": 18,
          "height": 380,
          "width": 320,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 0.3333333333333333,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 58,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "padding": 3,
          "width": 220,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#555",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 16,
            }
          }
        >
          Escolha uma foto
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "height": 58,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "padding": 3,
          "width": 220,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#555",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 16,
            }
          }
        >
          Use essa foto
        </Text>
      </View>
    </View>
  </View>
</View>
`;
